### PR TITLE
Added expandedHeight in flexibleSpace and flexibleSpaceBar

### LINKED ISSUE #22651: Docs for flexibleSpace should specify that expandedHeight is also needed in SliverAppBar
[flexibleSpace](https://docs.flutter.io/flutter/material/SliverAppBar/flexibleSpace.html) docs do not currently say that you also need to also specify the expandedHeight parameter to be able to work. (the expandedHeight parameter says that, but if you don't look there, you don't know)

Also possibly worth mentioning in the FlexibleSpaceBar [class docs](https://docs.flutter.io/flutter/material/FlexibleSpaceBar-class.html) as well.

### PR DESCRIPTION
**Pull request by Pragya007**
## Description

Added `expandedHeight` in `flexibleSpace` and `flexibleSpaceBar` API docs.

## Related Issues

Fixes [#22651](https://github.com/flutter/flutter/issues/22651)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. 

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/flexible_space_bar.dart [hunk 1]
```diff
@@ -41,11 +41,14 @@ enum StretchMode {
...
+/// Most commonly used in the [SliverAppBar.flexibleSpace] field, a flexible
 /// space bar expands and contracts as the app scrolls so that the [AppBar]
 /// reaches from the top of the app to the top of the scrolling contents of the
-/// app. Furthermore is included functionality for stretch behavior. When
-/// [SliverAppBar.stretch] is true, and your [ScrollPhysics] allow for
+/// app. Also, while including [SliverAppBar.flexibleSpace] it is important to
+/// mention [expandedHeight] property because if a [flexibleSpace] widget is
+/// specified [expandedHeight] should be big enough to accommodate whatever that
+/// widget contains. Furthermore is included functionality for stretch behavior.
+/// When [SliverAppBar.stretch] is true, and your [ScrollPhysics] allow for
```
Comment (Piinks) [inline packages/flutter/lib/src/material/flexible_space_bar.dart:51]:
```suggestion
/// app. When using [SliverAppBar.flexibleSpace], the [SliverAppBar.expandedHeight]
/// must be large enough to accommodate the [SliverAppBar.flexibleSpace] widget.
/// 
/// Furthermore is included functionality for stretch behavior. When
/// [SliverAppBar.stretch] is true, and your [ScrollPhysics] allow for
```

File: packages/flutter/lib/src/material/app_bar.dart [hunk 2]
```diff
@@ -322,6 +322,11 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
   /// changes the [AppBar]'s size. A [SliverAppBar] in a [CustomScrollView]
   /// changes the [AppBar]'s height when scrolled.
   ///
+  /// While including [SliverAppBar.flexibleSpace] it is important to mention
+  /// [expandedHeight] property because if a [flexibleSpace] widget is specified
+  /// [expandedHeight] should be big enough to accommodate whatever that widget
+  /// contains.
+  ///
```
Comment (Piinks) [inline packages/flutter/lib/src/material/app_bar.dart:329]: We can simplify this a bit, and let's actually move it to the 'flexibleSpace' of the `SliverAppBar` class where it is relevant.
The `AppBar` does not have an `expandedHeight`.
```suggestion
  /// When using [SliverAppBar.flexibleSpace], the [SliverAppBar.expandedHeight]
  /// must be large enough to accommodate the [SliverAppBar.flexibleSpace] widget.
  ///
```

**Review by Piinks — CHANGES_REQUESTED** [review]
Hey @Pragya007 thank you for making our documentation better! :)

File: packages/flutter/lib/src/material/app_bar.dart (hunk 2 above)
Comment (Pragya007) [reply packages/flutter/lib/src/material/app_bar.dart:329]: Sure @Piinks I will make the required changes.

Comment (Pragya007) [reply packages/flutter/lib/src/material/app_bar.dart:329]: Are you talking about [this](https://github.com/flutter/flutter/blob/1ad9baa8b99a2897c20f9e6e54d3b9b359ade314/packages/flutter/lib/src/material/app_bar.dart#L957), but it is present in [appBar](https://github.com/flutter/flutter/blob/1ad9baa8b99a2897c20f9e6e54d3b9b359ade314/packages/flutter/lib/src/material/app_bar.dart#L1056).
And I have changed the line for above, flexibleSpaceBar.

**Review by Piinks — APPROVED** [review]
LGTM! Thank you for the contribution!

**Pull request merged**